lib/GenerateResource.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * GenerateResource.js - Generate JSON resource only with xliff files.
+ *
+ * Copyright © 2020, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+var utils = require("./utils.js");
+var LocaleMatcher = require("ilib/lib/LocaleMatcher");
+var TranslationSet = require("./TranslationSet.js");
+var Xliff = require("./Xliff.js");
+
+var logger = log4js.getLogger("loctool.lib.GenerateResource");
+
+/**
+ * @class A class that represents the local story of a set of
+ * translations used in a project.
+ *
+ * @constructor
+ * @param {String} sourceLocale the source locale for this set
+ */
+var GenerateResource = function (options) {
+    logger.trace("GenerateResource constructor called");
+    this.targetDir = ".";
+    this.xliffsDir = ".";
+    this.resRoot = "resources";
+    this.resName = "strings.json";
+
+    if (options && options.xliffsDir) {
+        this.xliffsDir = options.xliffsDir;
+    }
+    if (options &&  options.resRoot) {
+        this.resRoot = options.resRoot;
+    }
+    if (options && options.resName) {
+        this.resName = options.resName;
+    }
+    this.ts = new TranslationSet(this.sourceLocale);
+};
+
+var xliffFileFilter = /(^|[^a-z])([a-z][a-z][a-z]?)(-([A-Z][a-z][a-z][a-z]))?(-([A-Z][A-Z]))?\.xliff$/;
```

Comment:
There may be a prefix to the xliff file name, like "myapp-ko-KR.xliff", so this should be updated